File: src/pocketcloud/cloud/network/packet/impl/response/ServerHandshakeResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response;

use pocketcloud\cloud\config\impl\MainConfig;
use pocketcloud\cloud\network\packet\data\PacketData;
use pocketcloud\cloud\network\packet\impl\type\VerifyStatus;
use pocketcloud\cloud\network\packet\ResponsePacket;

final class ServerHandshakeResponsePacket extends ResponsePacket {

    private string $prefix;

    public function __construct(
        private ?VerifyStatus $verifyStatus = null,
    ) {
        $this->prefix = MainConfig::getInstance()->getInGamePrefix();
    }

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeVerifyStatus($this->verifyStatus)
            ->write($this->prefix);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->verifyStatus = $packetData->readVerifyStatus();
        $this->prefix = $packetData->readString();
    }

    public function getVerifyStatus(): ?VerifyStatus {
        return $this->verifyStatus;
    }

    public function getPrefix(): string {
        return $this->prefix;
    }

    public static function create(VerifyStatus $verifyStatus): self {
        return new self($verifyStatus);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/response/ServerHandshakeResponsePacket.java
package de.pocketcloud.network.packet.impl.response;

import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.api.server.VerificationStatus;
import de.pocketcloud.network.packet.ResponsePacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ServerHandshakeResponsePacket extends ResponsePacket implements ClientboundPacket {

    private VerificationStatus verificationStatus;

    public ServerHandshakeResponsePacket(VerificationStatus verificationStatus) {
        this.verificationStatus = verificationStatus;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(verificationStatus);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static ServerHandshakeResponsePacket create(VerificationStatus verificationStatus) {
        return new ServerHandshakeResponsePacket(verificationStatus);
    }
}
